packit/base_git.py

```diff
@@ -351,21 +351,39 @@ def fetch_upstream_archive(self):
         with cwd(self.absolute_specfile_dir):
             self.download_remote_sources()
 
-    def set_specfile_content(self, specfile: Specfile, version: str, comment: str):
+    def set_specfile_content(
+        self,
+        specfile: Specfile,
+        version: str,
+        comment: str,
+        preserve_autochangelog: bool = False,
+    ):
         """
         update this specfile using provided specfile
 
         :param specfile: specfile to get changes from (we update self.specfile)
         :param version: version to set in self.specfile
         :param comment: new comment for the version in %changelog
+        :param preserve_autochangelog: preserve the %autochangelog macro
```

Comment:
I was a bit confused by the name of the argument at first. From the first usage (if at line `371`), it looks like `has_autochangelog` but the second occurrence (if at line `384`) looks more like an `add_new_changelog_entry`.

Reply:
You are right, it seems confusing.
I tried to refactor this but line 370 made me roll back all the changes.
If I understand properly the code, in line 370 the downstream spec file content is overwritten by the **original downstream** spec file content. And you already told me a couple of good reasons for doing it.

In line 371, I am preserving the upstream %autochangelog macro (if any in the **upstream spec file**). In the case, as an example, that the original downstream spec file does not have one. I think this could happen, am I wrong?

In line 384, the version and the changelog are updated, if any. You are right, it seems I add a new empty changelog entry because I need to add a new version. I can not split changing the version and changing the changelog. Because any change to the version will be potentially overwritten by line 370 called when changing the changelog... thus the version must be updated only after that the changelog is updated...

I am not sure I answered your concerns.  Please let me know If you have any suggestion on how to improve this :)

Reply:
I don't have anything about the implementation, but am just thinking if we can find a better name for that argument..;) But thanks for the detailed comment. Now I see why it's like this.

Maybe just more detailed parameter documentation will be enough. What about mentioning that when `preserve_autochangelog` is set, we sync the macro to downstream and add no changelog entry.

Reply:
Yes, sure